.internal/copyArrayWithOwnProperty.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * Copies the values of `source` to `array`.
+ *
+ * @private
+ * @param {Array} source The array to copy values from.
+ * @param {Array} [array=[]] The array to copy values to.
+ * @returns {Array} Returns `array`.
+ */
+function copyArrayWithOwnProperty(source, array) {
+  const length = source.length
```

Comment:
const length = source?.length ?? 0

Reply:
I fixed it, but I think it’s better to judge by Array.isArray

Reply:
What do you think?

Reply:
Yes, it seems to be more appropriate for the said scenario. And in case the source is not array we should throw an exception. Thoughts ?

Reply:
`!Array.isArray(source) && throw new Error('source is not an array!') ;`

Reply:
Now , in the suggested workflow , it will throw an exception if the supplied argument is not an array.